### ISSUE TITLE
Current logic in the frontend prevents using the default filename

### ISSUE THREAD

**Issue by markelg** [pangeo-data/xESMF#153]
Current logic in the frontend does not let me to use the default filename

```
        # some logic about reusing weights with either filename or weights args
        if reuse_weights and (filename is None) and (weights is None):
            raise ValueError('To reuse weights, you need to provide either filename or weights.')
```

I think it should try to load the weights from default filename instead of raising an error. Am I missing something?

The pattern used to build the default filename is good and useful. I can find some time to write a PR if you want.

**Comment by raphaeldussin** [comment]
agreed, I think something in the logic has been broken. @huard and @aulemahal any thoughts on this?
One workaround is for you to specify explicitly the name of the weights file you want to write.

PRs are always welcome!

**Comment by aulemahal** [comment]
I agree that this would make sense! I don't think it ever worked, though, seems to me that the current behaviour was introduced when `reuse_weights` was introduced, but I might be mistaken.

It think it would be a matter of setting `self.filename` prior to this check and then add a check to see if the file exists.

**Comment by huard** [comment]
The filename logic dates back to the time ESMF wrote the weights to file, and xESMF had to read them from disk. About two years ago, ESMF introduced a mechanism to share the weights in memory. The current code is the result of trying to support both the older file-based and memory-based approaches without breaking legacy code.

I'm not sure I would want xESMF to automatically read from a default file name that is not directly visible from the docstring. It feels error-prone.
